### PR TITLE
Better random input for dynamic shape

### PR DESCRIPTION
**Pull request by keehyuna**
# Description

Random bool/integer type input for dynamic shape is casted from random float in [0,1)
All zero input might cause false positive test result.

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.
- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/_Input.py [hunk 1]
```diff
@@ -352,9 +353,20 @@ def example_tensor(
                 )
             else:
                 if isinstance(self.shape, tuple):
-                    return torch.rand(self.shape).to(
-                        dtype=self.dtype.to(torch.dtype, use_default=True)
-                    )
+                    if self.dtype in [dtype.u8, dtype.i8, dtype.i32, dtype.i64]:
```
Comment (chohk88) [inline py/torch_tensorrt/_Input.py:356]: It appears that the changes ensure values are within the range of -128 to 127 for `dtype.i8`, `dtype.i32`, and `dtype.i64`, and within the range of 0 to 127 for `dtype.u8`. 

Previously, it seems there was an issue where casting random values between 0 and 1 to integers resulted in most values being 0.

This fix seems necessary and addresses the problem effectively.

File: tests/py/dynamo/conversion/harness.py [hunk 2]
```diff
@@ -353,6 +353,15 @@ def run_test_compare_tensor_attributes_only(
             debug=True,
         )
 
+        output_dtypes = None
+        if check_dtype:
```
Comment (chohk88) [inline tests/py/dynamo/conversion/harness.py:357]: I believe your modification is correct, but I'm curious about the history. Why was it originally necessary to specify `output_dtypes` as an input in the `dynamic shape test` function? It seems that specifying `output_dtypes` is no longer needed, and your modification looks like the right approach.

**Review by chohk88 — APPROVED** [review]
LGTM, but I didn't know the history of `output_dtypes` so I left a comment about it.

File: tests/py/dynamo/conversion/harness.py (hunk 2 above)
Comment (keehyuna) [reply tests/py/dynamo/conversion/harness.py]: Thanks, I agree. it was included from other issue.

File: py/torch_tensorrt/_Input.py (hunk 1 above)
Comment (keehyuna) [reply py/torch_tensorrt/_Input.py]: Yes, but It caused problem in model test as random int value was used in fake tensor as index.
Added option to initialize it with zero.

File: py/torch_tensorrt/_Input.py [hunk 3]
```diff
@@ -183,9 +183,9 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
                     kwargs["opt_shape"], dtype=kwargs["dtype"]
                 )
             elif self.shape_mode == Input._ShapeMode.DYNAMIC:
-                self.torch_tensor = self.example_tensor("opt_shape")
+                self.torch_tensor = self.example_tensor("opt_shape", zero_filled=True)
             else:
-                self.torch_tensor = self.example_tensor()
+                self.torch_tensor = self.example_tensor(zero_filled=True)
```
Comment (keehyuna) [inline py/torch_tensorrt/_Input.py:188]: If input is used as fake tensor and used as index, it caused cuda error. Input buffer is initialized as zero.

File: py/torch_tensorrt/_Input.py [hunk 4]
```diff
@@ -384,4 +384,21 @@ def example_tensor(
                 raise ValueError(
                     "Requested an example tensor from a dynamic shaped input but did not specific which profile field to use."
                 )
+        type = self.dtype.to(torch.dtype, use_default=True)
+        if zero_filled:
+            return torch.zeros(shape, dtype=type)
+        else:
+            if self.dtype in [dtype.u8, dtype.i8, dtype.i32, dtype.i64]:
+                return torch.randint(
+                    max(torch.iinfo(torch.int8).min, torch.iinfo(type).min),
```
Comment (keehyuna) [inline py/torch_tensorrt/_Input.py:393]: unary like ACOS does not support int32 in the TRT, but supports int8. Using 8bit range random integer to cover all int type.
